### DISCUSSION TITLE
Support - Can't upgrade past 2.0.1 because of build errors

### DISCUSSION THREAD

**Discussion by SquaredSee** [nathanwoulfe/Plumber-2#116] [Q&A] [answered]
Hello,

I am running Umbraco 9, I know that plumber still only officially supports Umbraco 8, but if you have any insights they would be appreciated. 

I am having the same problem as the user in issue #108, where you responded that it is fixed in 2.0.2. I am attempting to update to 2.0.2 (I have also tried 2.1.0), however I am getting the following build errors:
`D:\Users\ccrouch\.nuget\packages\plumber.backoffice\2.0.2\buildTransitive\Plumber.Backoffice.targets(77,5): error MSB4801: The task factory "CodeTaskFactory" is not supported on the .NET Core version of MSBuild. [D:\Users\ccrouch\Documents\code\GPSCommsPortal\GPSCommsPortal.csproj]D:\Users\ccrouch\.nuget\packages\plumber.backoffice\2.0.2\buildTransitive\Plumber.Backoffice.targets(77,5): error MSB4175: The task factory "CodeTaskFactory" could not be loaded from the assembly "C:\Program Files\dotnet\sdk\5.0.404\Microsoft.Build.Tasks.Core.dll". The task factory must return a value for the "TaskType" property. [D:\Users\ccrouch\Documents\code\GPSCommsPortal\GPSCommsPortal.csproj]`

I've gathered that this probably has to do with me running Umbraco 9. Is there anything that can be done about this, or are we out of luck? Thanks!

**Comment by nathanwoulfe** [comment, marked as answer]
Hey @SquaredSee, Plumber2 will absolutely run on Umbraco9, it's all fully cross-platform, multi-targeted NetCore friendly.

Issue looks like this one - https://github.com/dotnet/msbuild/issues/616 - so I think I'll need to update the targets plus add backwards compat for older msbuild (ie CodeTaskFactory rather than RoslynCodeTaskFactory).

**Reply by nathanwoulfe** [reply]
This will be fixed in 2.1.1, there's an RC building now, will be on Nuget shortly if you're able to test and confirm the fix. Looks to be fine, I installed it into a v9 site, ran `dotnet build` and worked fine.

**Reply by SquaredSee** [reply]
That worked, thanks for the help!

**Reply by nathanwoulfe** [reply]
Happy days! I'll fire off a proper release today sometime

**Comment by nathanwoulfe** [comment]
v2.1.1 is available now, fixes this issue 👍
